### DISCUSSION TITLE
How to add revalidation for components like Header, Footer?

### DISCUSSION THREAD

**Discussion by stepan-twnty** [vercel/next.js#66400] [Help]
### Summary

I have a lot of pages ~1000 and need to keep my two header components updated after some changes on CMS. 

What do I have now? After the changes on CMS, I rebuilt the whole site. Before it worked OK because the build time was less than 5 min but after some business requirements my build time is 80 min.

Making fetch on the client's side is not a good idea due to [CLS](https://web.dev/articles/cls?hl=ru).

I have only one idea it's created for each header a separate page e.g. `pages/header_1` and `pages/header_2` and then put these like iframes `<iframe src="/header_1" /> for some pages where I need it. I've checked it and looks like it works but there are some problems with styling and global states (states are different for iframe in comparing with a page)

Would someone happen to have experience with similar and can help with a better idea?

### Additional information

I don't use app folders only pages.
Add revalidation for each page or on demand I don't think it's good idea.

**Comment by shubhamw3nuts** [comment]
I am also facing a similar issue can anyone help on how to revalidate header and footer components?
